src/html_tag.cpp

```diff
@@ -2751,10 +2751,15 @@ litehtml::element::ptr litehtml::html_tag::get_child( int idx ) const
 
 size_t litehtml::html_tag::get_index() const
 {
+	return get_index( this );
+}
+
+size_t litehtml::html_tag::get_index( const litehtml::element * element ) const
+{

```

Comment:
I think you misread the definition. In case an object is passed, index() returns the index of the object inside the collection, not inside the DOM. So this is a jquery level, not in the litehtml layer